src/modules/fw_pos_control_l1/fw_pos_control_l1_main.cpp

```diff
@@ -2151,11 +2153,27 @@ void FixedwingPositionControl::tecs_update_pitch_throttle(float alt_sp, float v_
 		const math::Vector<3> &ground_speed,
 		unsigned mode, bool pitch_max_special)
 {
-	/* do not run tecs if we are not in air */
-	if (_vehicle_status.condition_landed) {
+	bool run_tecs = true;
+
+	// do not run TECS if we are not in air
+	run_tecs &= !_vehicle_status.condition_landed;
+
+	// do not run TECS if vehicle is a VTOL and we are in rotary wing mode
+	if (_vehicle_status.is_vtol) {
+		run_tecs &= !_vehicle_status.is_rotary_wing;
```

Comment:
Will have to test that. I don't expect TECS to do anything special during the transition.